src/xutils.cpp

```diff
@@ -149,4 +149,38 @@ namespace xpyt
 
         std::clog << "PYTHONHOME set to " << mbstr << std::endl;
     }
+
+    // Parse a version string to vector of int.
+    // The string must be formatted as the regex: [0-9]+(\.[0-9]+)*
+    std::vector<int> parse_version(std::string version)
+    {
+        int value;
+        std::vector<int> result;
+        std::istringstream parser(version);
+
+        parser >> value;
+        result.push_back(value);
+        parser.get();
+        while(parser)
+        {
+            parser >> value;
+            result.push_back(value);
+            parser.get();
+        }
+        return result;
+    }
+
+    // Compare 2 version and return true if version1 < version2.
+    // The versions must be strings formatted as the regex: [0-9]+(\.[0-9]+)*
+    bool less_than_version(std::string version1, std::string version2)
+    {
```

Comment:
If we assume that `version1` and `version2` have the format "MAJOR MINOR PATCH", then the implementation of this function can be improved:

```cpp
using iterator_type = std::istream_iterator<int>;
std::istringstream iv1(version1), iv2(version2);
return std::lexicographical_compare(
    iterator_type(iv1),
    iterator_type(),
    iterator_type(iv2),
    iterator_type()
);
```

Reply:
Thanks @JohanMabille.
Not sure we can assume it as users can still have alpha/beta versions I thinks.

Reply:
In that case we need to split the string to remove the alpha / beta part.

Reply:
OK, let's do it this way